store/store.go

```diff
@@ -158,6 +158,12 @@ type Store struct {
 	proxyConnectHeader http.Header
 
 	userAgent string
+
+	xdeltaCheckLock sync.Mutex
+	// whether we should use deltas or not
+	shouldUseDeltas *bool
+	// which xdelta3 we picked when we checked the deltas
+	xdelta3CmdLocation string
```

Comment:
it sounds like this one should become a callback that can be set to return either the a exec.Cmd for the host or for system snap xdelta3:

xdelta3Cmd  func()  (*exec.Cmd, error)